lib/renderer.js

```diff
@@ -1152,10 +1174,17 @@
             hasZeroBounds = layer && (!layer.bounds || (layer.bounds.top === 0 &&
                                                         layer.bounds.bottom === 0 &&
                                                         layer.bounds.left === 0 &&
-                                                        layer.bounds.right === 0));
+                                                        layer.bounds.right === 0)),
+            documentHasZeroBounds = document && this._documentHasZeroBounds(document);
```

Comment:
You should make `_documentHasZeroBounds` more generic and also use it for calculating `hasZeroBounds` for layer.